hazelcast/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -170,25 +170,18 @@
     private final HazelcastInstance instance;
     private final ILogger logger;
 
-    private final IMap<Long, JobRecord> jobRecords;
-    private final IMap<Long, JobExecutionRecord> jobExecutionRecords;
-    private final IMap<Long, JobResult> jobResults;
-    private final IMap<Long, List<RawJobMetrics>> jobMetrics;
-    private final IMap<String, SnapshotValidationRecord> exportedSnapshotDetailsCache;
-    private final FlakeIdGenerator idGenerator;
+    private IMap<Long, JobRecord> jobRecords;
+    private IMap<Long, JobExecutionRecord> jobExecutionRecords;
+    private IMap<Long, JobResult> jobResults;
```

Comment:
Isn't it possible here that we will see half-initialized IMap instance? E.g. one thread is initializing this field, another thread reads the IMap reference, but it's still being initialized.

Reply:
should be fixed with `ConcurrentMemoizingSupplier` changes